File: C++ 28tech T7_2024/contest_2/1.cpp
// bài 1 tính toán giá trị biểu thức.

//cho biểu thức A(x) = X^3 + 3X^2 + x + 1. với giá trị của x được nhập từ bàn phím,tính và in ra giá trị của biểu thức trên 

//Đầu Vào:
//Số nguyên x

//Rằng Buộc
//-10^5 <= n <= 10^5

// Đầu ra 
//in ra kết quả của biểu thức

//ví dụ 
//input1
//2
//output1
//23
#include <iostream> 
using namespace std;

int main(){
	int n;cin >> n;
	
	long long result = (long long)n * n * n + (long long) 3 * n * n + n + 1;
	cout << result << endl;
	
}

File: Khác/hsg_2024/đề 4/1.py
# Bài 1. Viết chương trình cho phép người dùng nhập một số nguyên dương N từ bàn phím rồi in ra số có nhiều ước số nhất trong các số nhỏ hơn N


def uoc(n):
    count = 0
    for i in range(1,n+1):
        if n % i == 0:
            count += 1
    return count

def main(N):
    max_ = 0
    result = 1
    for i in range(1,N):
        n = uoc(i)
        if n > max_:
            max_ = n 
            result = i

    return result,max_

# n = input("nhập N: ")
n = 10
if n <= 1:print("n phải lớn hơn 1!")
else: print("số có nhiều ước nhất là:",main(n)[0],"và có số ước là:",main(n)[-1])
